### PR TITLE
3269 configure default library view

### LINKED ISSUE #3269: Allow Table View to be configured as "default" library view in settings

### PR DESCRIPTION
**Pull request by fnocetti**
fixes #3269 

PR checklist:
- [ ] Update READ.me ?
- [ ] Update API documentation ?

QA checklist:
- [ ] Smoke test the functionality described in the issue
- [ ] Test for side effects
- [ ] UI responsiveness
- [ ] Cross browser testing
- [ ] Code review

### CODE REVIEW

**Comment by daneryl** [comment]
@fnocetti when i load the root page, localhost:3000 it still loads the cards view as default even if i have table or map configured, is this the intended behavior ?

**Comment by fnocetti** [comment]
> @fnocetti when i load the root page, localhost:3000 it still loads the cards view as default even if i have table or map configured, is this the intended behavior ?

@daneryl My understanding is that yes, this is the intended behavior: only change what the "public documents" button does. If the user needs the root page to show a table view they can easily use /library/table as a custom home page.

File: app/react/App/Menu.js [hunk 1]
```diff
@@ -12,12 +12,29 @@ import { showSemanticSearch } from 'app/SemanticSearch/actions/actions';
 import { FeatureToggleSemanticSearch } from 'app/SemanticSearch/components/FeatureToggleSemanticSearch';
 import { Icon } from 'UI';
 
+const defaultLibraryOptionsToInfo = {
```
[the remark this answers is not in the document]
Comment (fnocetti) [reply app/react/App/Menu.js]: Sure!

**Comment by fnocetti** [comment]
@mfacar @daneryl thanks! After our chat today I implemented the behavior for the home page too. Please note that I had to refactor a bit of the custom home page code on the routes. I checked as many scenarios as I could imagine, but it will be nice if you can test it too. The refactor is isolated to the penultimate commit "Mini refactor of the home page route" so you can review its changes.

**Review by mfacar — APPROVED** [review]
LGTM, Great job!

**Pull request merged**